raft/src/main/java/org/apache/kafka/snapshot/SnapshotWriter.java

```diff
@@ -84,44 +74,36 @@ private SnapshotWriter(
     }
 
     /**
-     * Adds a {@link MetadataSnapshotHeaderRecord} to snapshot
+     * Adds a {@link SnapshotHeaderRecord} to snapshot
      *
      * @throws IllegalStateException if the snapshot is not empty
      */
     private void initializeSnapshotWithHeader() {
         if (snapshot.sizeInBytes() != 0) {
-            throw new IllegalStateException("Initializing new snapshot (ID: "
-                    + snapshot.snapshotId().epoch + ", " + snapshot.snapshotId().offset
-                    + ") with a non-empty file");
+            throw new IllegalStateException("Initializing with a non-empty snapshot (%s)"
+                    + snapshot.snapshotId());
```

Comment:
`+` does a string concatenation. You need to use `String.format` if you wan to perform string formatting.